common/test/unit/com/thoughtworks/go/domain/PipelineConfigTest.java

```diff
@@ -236,15 +239,71 @@ public void shouldGetAllTemplateVariableNames() {
+        git.setName(new CaseInsensitiveString("git"));
+
+        PipelineConfig pipelineConfig = new PipelineConfig(new CaseInsensitiveString("cruise"), new MaterialConfigs(git));
+        pipelineConfig.setLabelTemplate(labelFormat);
+
+        pipelineConfig.validate(null);
+
+        return pipelineConfig;
+    }
+
+    @Test
+    public void shouldValidateCorrectPipelineLabelWithoutTruncationSyntax() {
+        String labelFormat = "pipeline-${COUNT}-${git}-454";
+        PipelineConfig pipelineConfig = createAndValidatePipelineLabel(labelFormat);
+        assertThat(pipelineConfig.errors().on(PipelineConfig.LABEL_TEMPLATE), is(nullValue()));
+    }
+
+    @Test
+    public void shouldValidatePipelineLabelWithNonExistingMaterial() {
+        String labelFormat = "pipeline-${COUNT}-${NoSuchMaterial}";
+        PipelineConfig pipelineConfig = createAndValidatePipelineLabel(labelFormat);
+        assertThat(pipelineConfig.errors().on(PipelineConfig.LABEL_TEMPLATE), startsWith("You have defined a label template in pipeline"));
+    }
+
+    @Test
+    public void shouldValidateCorrectPipelineLabelWithTruncationSyntax() {
+        String labelFormat = "pipeline-${COUNT}-${git[:7]}-alpha";
+        PipelineConfig pipelineConfig = createAndValidatePipelineLabel(labelFormat);
+        assertThat(pipelineConfig.errors().on(PipelineConfig.LABEL_TEMPLATE), is(nullValue()));
+    }
+
+    @Test
+    public void shouldValidateIncorrectPipelineLabelWithTruncationSyntax() {
+        String labelFormat = "pipeline-${COUNT}-${noSuch[:7]}-alpha";
+        PipelineConfig pipelineConfig = createAndValidatePipelineLabel(labelFormat);
+        assertThat(pipelineConfig.errors().on(PipelineConfig.LABEL_TEMPLATE), startsWith("You have defined a label template in pipeline"));
+    }
+
+    @Test
+    public void shouldSupportTruncationSyntax() {
+        PipelineConfig pipelineConfig = new PipelineConfig(new CaseInsensitiveString("cruise"), new MaterialConfigs());
+        pipelineConfig.setLabelTemplate("pipeline-${COUNT}-${git[:7]}-alpha");
+
+        Set<String> variables = pipelineConfig.getTemplateVariables();
+        assertThat(variables, contains("COUNT", "git"));
+        assertThat(variables.size(), is(2));
+    }
+
+    @Test
     public void shouldSupportSpecialCharactors() {
         PipelineConfig pipelineConfig = new PipelineConfig(new CaseInsensitiveString("cruise"), new MaterialConfigs());
         pipelineConfig.setLabelTemplate("pipeline-${COUN_T}-${my-material}${h.i}${**}");
 
         Set<String> variables = pipelineConfig.getTemplateVariables();
-        assertThat(variables.contains("COUN_T"), is(true));
-        assertThat(variables.contains("my-material"), is(true));
-        assertThat(variables.contains("h.i"), is(true));
-        assertThat(variables.contains("**"), is(true));
+        assertThat(variables, contains("COUN_T", "my-material", "h.i", "**"));
```

Comment:
I prefer the use of `assertThat(list, contains("elem1", "elem2")` as this yields error message which are more expressive than the one for `assertThat(list.contains("elem1"), is(true)); ...`